src/dds_data.cpp

```diff
@@ -209,8 +207,223 @@ QVariant CommonData::readValue(const QString& topicName,
+          }
+          break;
+      }
+    case OpenDDS::XTypes::TK_INT64:
+      {
+          CORBA::LongLong tmp;
+          rc = parent_data->get_int64_value(tmp, id);
+          if (rc == DDS::RETCODE_OK) {
+              value = static_cast<qlonglong>(tmp);
+              failed = false;
+          }
+          break;
+      }
+    case OpenDDS::XTypes::TK_UINT64:
+      {
+          CORBA::ULongLong tmp;
+          rc = parent_data->get_uint64_value(tmp, id);
+          if (rc == DDS::RETCODE_OK) {
+              value = static_cast<qulonglong>(tmp);
+              failed = false;
+          }
+          break;
+      }
+    case OpenDDS::XTypes::TK_FLOAT32:
+      {
+          CORBA::Float tmp;
+          rc = parent_data->get_float32_value(tmp, id);
+          if (rc == DDS::RETCODE_OK) {
+              value = tmp;
+              failed = false;
+          }
+          break;
+      }
+    case OpenDDS::XTypes::TK_FLOAT64:
+      {
+          CORBA::Double tmp;
+          rc = parent_data->get_float64_value(tmp, id);
+          if (rc == DDS::RETCODE_OK) {
+              value = tmp;
+              failed = false;
+          }
+          break;
+      }
+    case OpenDDS::XTypes::TK_CHAR8:
+      {
+          CORBA::Char tmp;
+          rc = parent_data->get_char8_value(tmp, id);
+          if (rc == DDS::RETCODE_OK) {
+              value = tmp;
+              failed = false;
+          }
+          break;
+      }
+    case OpenDDS::XTypes::TK_CHAR16:
+      {
+          CORBA::WChar tmp;
+          rc = parent_data->get_char16_value(tmp, id);
+          if (rc == DDS::RETCODE_OK) {
+              value = tmp;
+              failed = false;
+          }
+          break;
+      }
+    case OpenDDS::XTypes::TK_STRING8:
+      {
+          CORBA::String_var tmp;
+          rc = parent_data->get_string_value(tmp, id);
+          if (rc == DDS::RETCODE_OK) {
+              value = tmp.in();
+              failed = false;
+          }
+          break;
+      }
+    case OpenDDS::XTypes::TK_ENUM:
+      {
+          CORBA::Long tmp;
+          rc = parent_data->get_int32_value(tmp, id);
+          if (rc == DDS::RETCODE_OK) {
+              DDS::String8_var name;
+              rc = OpenDDS::XTypes::get_enumerator_name(name, tmp, md->type());
+              if (rc == DDS::RETCODE_OK) {
+                  value = name.in();
+                  failed = false;
+              }
+          }
+          break;
+      }
+    default:
+        break;
+    }
+
+    if (failed) {
+        value = error;
+    }
+    return value;
+}
 
-} // End CommonData::readValue
+//------------------------------------------------------------------------------
+QVariant CommonData::readValue(const QString& topicName,
+                               const QString& memberName,
+                               unsigned int index)
+{
+    const QVariant value = readMember(topicName, memberName, index);
+    if (value.toString() != "NULL") {
```

Comment:
Would `value.isValid()` or `isNull()` work here?

Reply:
That's true. It looks like a separate problem. Should we fix that in this PR or are you okay with a new PR for that?

Reply:
Consider this potential change in this PR: `readDynamicMember()` returns a null variant on error, but the UI will never see that.  Instead it's used internally to switch to `readMember()` to get a value for the UI.

Reply:
That sounds good. I've updated.